packages/core-data/src/selectors.js

```diff
@@ -168,3 +168,36 @@ export const getEntityRecords = createSelector(
 export function getThemeSupports( state ) {
 	return state.themeSupports;
 }
+
+/**
+ * Returns the embed preview for the given URL.
+ *
+ * @param {Object} state    Data state.
+ * @param {string} url      Embedded URL.
+ *
+ * @return {*} Undefined if the preview has not been fetched, otherwise, the preview fetched from the embed preview API.
+ */
+export function getEmbedPreview( state, url ) {
+	return state.embedPreviews[ url ];
+}
+
+/**
+ * Determines if the returned preview is an oEmbed link fallback.
+ *
+ * WordPress can be configured to return a simple link to a URL if it is not embeddable.
+ * We need to be able to determine if a URL is embeddable or not, based on what we
+ * get back from the oEmbed preview API.
+ *
+ * @param {Object} state    Data state.
+ * @param {string} url      Embedded URL.
+ *
+ * @return {booleans} Is the preview for the URL an oEmbed link fallback.
+ */
+export function isPreviewEmbedFallback( state, url ) {
+	const preview = state.embedPreviews[ url ];
+	const oEmbedLinkCheck = '<a href="' + url + '">' + url + '</a>';
+	if ( ! preview ) {
+		return false;
+	}
+	return preview.html === oEmbedLinkCheck;
```

Comment:
It seems fragile to test the `html`. WP can decide to change the fallback right? Can't we add an explicit attribute to the embed endpoint when it's a fallback: `isFallback: true`?

Reply:
The embed fallbacks are currently done by a filter in `embed_maybe_make_link`, and that _only_ deals with HTML, you can't change any of the data structure. We'd have to change the core embed code itself to do that.